.github/workflows/compile_protos.yml

```diff
@@ -5,111 +5,88 @@ concurrency:
+    runs-on: ubuntu-latest
+    steps:
+    - name: remove label
+      env:
+        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        GH_REPO: ${{ github.repository }}
+      run: gh pr edit ${{ github.event.pull_request.number }} --remove-label protos-compiled
+
   compile-protos:
-    if: >
-      github.event.workflow_run.event == 'pull_request_review' &&
-      github.event.workflow_run.conclusion == 'success'
-    runs-on: [self-hosted, x64]
-    container:
-      image: ghcr.io/viamrobotics/canon:amd64-cache
-      options: --platform linux/amd64
+    # run after unlabel so that ordering of remove-label, add-label is consistent
+    needs: unlabel
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') && contains(github.event.pull_request.labels.*.name, 'ready-for-protos')
+    runs-on: ubuntu-latest
     env:
       CI_COMMIT_MESSAGE_PREFIX: Built new protos from
       CI_COMMIT_AUTHOR: github-actions
-      LABEL_NAME: protos-compiled
     steps:
-      # Download PR info from approval workflow and checkout branch
-      - name: Download artifact
-        uses: dawidd6/action-download-artifact@v2
-        with:
-          run_id: ${{github.event.workflow_run.id }}
-      - name: Get PR Repo
-        id: get-pr-repo
-        uses: actions/github-script@v6
+      - uses: actions/checkout@v4
         with:
-          result-encoding: string
-          script: |
-            var fs = require('fs');
-            var repo = String(fs.readFileSync('./pr/repo')).trim();
-            return repo
-      - name: Get PR Ref
-        id: get-pr-ref
-        uses: actions/github-script@v6
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: actions/setup-node@v4
+      - uses: actions/setup-go@v5
         with:
-          result-encoding: string
-          script: |
-            var fs = require('fs');
-            var ref = String(fs.readFileSync('./pr/ref')).trim();
-            return ref
-      - name: Get PR Number
-        id: get-pr-number
-        uses: actions/github-script@v6
-        with:
-          script: |
-            var fs = require('fs');
-            var number = Number(String(fs.readFileSync('./pr/number')).trim());
-            return number
-      - uses: actions/checkout@v3
-        with:
-          repository: ${{ steps.get-pr-repo.outputs.result }}
-          ref: ${{ steps.get-pr-ref.outputs.result }}
-          token: ${{ secrets.REPO_READ_TOKEN }}
-
+          go-version: '1.21'
+      # todo: cache download
+      - name: grpc-web
+        run: |
+          wget --quiet https://github.com/grpc/grpc-web/releases/download/${{ env.GRPC_WEB_VERSION }}/protoc-gen-grpc-web-${{ env.GRPC_WEB_VERSION }}-linux-x86_64 -O protoc-gen-grpc-web
+          chmod +x protoc-gen-grpc-web
+          sudo mv protoc-gen-grpc-web /usr/local/bin
+          which protoc-gen-grpc-web
       # Set environment variables based on the last commit
       - name: Set environment variable "commit-message"
```

Comment:
Maybe `Set environment variables "commit-message" and "commit-author"` since we've consolidated this

Reply:
changed in 3ca0ef3